Makefile

```diff
@@ -0,0 +1,363 @@
+RABBITMQ_TOPOLOGY_OPERATOR_VERSION ?= 1.2.1
+install-rabbitmq-topology-operator: | install-cert-manager $(KUBECTL) ## Install RabbitMQ Topology Operator
+	$(KUBECTL) $(K_CMD) --filename \
+		https://github.com/rabbitmq/messaging-topology-operator/releases/download/v$(RABBITMQ_TOPOLOGY_OPERATOR_VERSION)/messaging-topology-operator-with-certmanager.yaml
+
+# https://github.com/knative/eventing/releases
+KNATIVE_EVENTING_VERSION ?= 1.0.0
+install-knative-eventing: | $(KUBECONFIG) $(KUBECTL) ## Install Knative Eventing
+	$(KUBECTL) $(K_CMD) --filename \
+		https://github.com/knative/eventing/releases/download/knative-v$(KNATIVE_EVENTING_VERSION)/eventing-crds.yaml
+	$(KUBECTL) $(K_CMD) --filename \
+		https://github.com/knative/eventing/releases/download/knative-v$(KNATIVE_EVENTING_VERSION)/eventing-core.yaml
+	$(KUBECTL) wait --for=condition=available deploy/eventing-controller --timeout=30s --namespace $(SYSTEM_NAMESPACE)
+	$(KUBECTL) wait --for=condition=available deploy/eventing-webhook --timeout=30s --namespace $(SYSTEM_NAMESPACE)
+
+install-knative-eventing-rabbitmq: | $(KUBECONFIG) $(KO) install-knative-eventing ## Install dev Knative Eventing RabbitMQ - also installs Knative Eventing
+	$(KO) apply --filename config/broker
+	$(KUBECTL) wait --for=condition=available deploy/rabbitmq-broker-controller --timeout=30s --namespace $(SYSTEM_NAMESPACE)
+	$(KUBECTL) wait --for=condition=available deploy/rabbitmq-broker-webhook --timeout=30s --namespace $(SYSTEM_NAMESPACE)
+	$(KO) apply --filename config/source
+	$(KUBECTL) wait --for=condition=available deploy/pingsource-mt-adapter --timeout=30s --namespace knative-eventing
+
+.PHONY: e2e-setup
+e2e-setup: install-rabbitmq-cluster-operator install-rabbitmq-topology-operator install-knative-eventing-rabbitmq ## Setup environment for e2e testing
+	@printf "$(INFO)Starting point: $(BOLD).github/workflows/kind-e2e.yaml$(RESET)\n"
+	@printf "$(INFO)This is slow - $(BOLD)33s cached$(RESET)$(INFO) - so it is kept as a non-explicit dependency for any e2e test$(RESET)\n"
+
+.PHONY: e2e-setup-rm
+e2e-setup-rm: | $(KIND)
+	$(KIND) delete cluster --name $(KIND_CLUSTER_NAME)
+
+.PHONY: test-e2e-publish
+test-e2e-publish: ## Run TestKoPublish end-to-end tests
+	@printf "$(WARN)Remember to run $(BOLD)make e2e-setup$(RESET)$(WARN) if needed$(RESET)\n"
+	go test -v -race -count=1 -timeout=15m -tags=e2e ./test/e2e/... -run 'TestKoPublish' \
+	| grep -v "no test files"
+
+.PHONY: test-e2e-broker
+test-e2e-broker: ## Run Broker end-to-end tests
+	@printf "$(WARN)Remember to run $(BOLD)make e2e-setup$(RESET)$(WARN) if needed$(RESET)\n"
+	@printf "$(WARN)$(BOLD)rabbitmqcluster$(RESET)$(WARN) has large resource requirements 🤔$(RESET)\n"
+	go test -v -race -count=1 -timeout=15m -tags=e2e ./test/e2e/... -run 'Test.*Broker.*' \
+	| grep -v "no test files"
+
+.PHONY: test-e2e-source
+test-e2e-source: ## Run Source end-to-end tests
+	@printf "$(WARN)Remember to run $(BOLD)make e2e-setup$(RESET)$(WARN) if needed$(RESET)\n"
```

Comment:
Telling someone to do something "if needed" is kind of roundabout, how do I know if it's needed?

Reply:
You are right, this makes more sense: https://github.com/knative-sandbox/eventing-rabbitmq/pull/525/commits/b05b259667e92edb460060e155d0651bcc30f428

How do you think that we should hint to the fact `KUBECONFIG` can be set in the environment, and then no kind cluster will be created?

Reply:
I think we should have kind cluster creation as an explicit step, not automatic/implicit

Reply:
I can see how it would make sense to fail if `KUBECONFIG` is not set, rather than default to creating a KiND cluster. Are you thinking the same thing?

Reply:
No, it's possible to not have `KUBECONFIG` set and still be valid, for example if you have a `~/.kube/config` file.